google/cloud/exceptions.py

```diff
@@ -41,7 +43,10 @@ def __init__(self, message, errors=()):
         self._errors = errors
 
     def __str__(self):
-        return '%d %s' % (self.code, self.message)
+        result = u'%d %s' % (self.code, self.message)
+        if six.PY2:
+            result = _to_bytes(result)
```

Comment:
@daspecster You'll still have a branch miss for the implicit `else: pass` branch. You need the pragma OR a unit test which monkey patches to make `six.PY2` false.